reports/_model.qmd

```diff
@@ -242,6 +242,222 @@ for (township in townships_to_iterate) {
+      triad_code == run_triad_code,
+      geography_type == "township_code",
+      by_class == TRUE
+    ) %>%
+    select(
+      geography_type,
+      geography_id,
+      class,
+      sale_fmv_q25,
+      sale_fmv_median,
+      sale_fmv_q75,
+      estimate_fmv_q25,
+      estimate_fmv_median,
+      estimate_fmv_q75
+    )
+
+  reshaped_data <- sales_vs_estimates_by_class %>%
+    mutate(township = ccao::town_convert(geography_id)) %>%
+    dplyr::filter(township == township_to_graph) %>%
+    pivot_longer(
+      cols = -c(geography_type, geography_id, township, class),
+      names_to = "type",
+      values_to = "value"
+    ) %>%
+    mutate(
+      category = ifelse(
+        grepl("sale", type),
+        paste(class, " - Sale"),
+        paste(class, " - Estimate")
+      )
+    ) %>%
+    select(category, value) %>%
+    mutate(
+      space_maker = as.numeric(factor(category)),
+      space_maker = space_maker + (space_maker - 1) %/% 2
+    )
+
+  line_ends <- reshaped_data %>%
+    group_by(space_maker, category) %>%
+    summarize(Start = min(value), End = max(value)) %>%
+    mutate(
+      color_class = ifelse(grepl("Sale", category), "Sale", "Estimate"),
+      # Set for legend re-ordering
+      color_class = factor(color_class, levels = c("Sale", "Estimate"))
+    )
+
+  print(
+    ggplot() +
+      geom_segment(
+        data = line_ends,
+        aes(
+          x = Start,
+          xend = End,
+          y = space_maker,
+          yend = space_maker,
+          color = color_class
+        ),
+        size = 1.2
+      ) +
+      geom_point(
+        data = reshaped_data,
+        aes(
+          x = value, y = space_maker
+        ),
+        color = "black",
+        size = 2,
+        shape = 21,
+        fill = "white"
+      ) +
+      theme_minimal() +
+      theme(
+        axis.title.x = element_text(color = "darkblue", size = 12),
+        axis.title.y = element_text(color = "darkblue", size = 12),
+        legend.position = c(.9, .9)
+      ) +
+      labs(
+        x = "Value",
+        y = "Class",
+        title = "Sales vs Estimates by Township",
+        subtitle = "25th percentile, median, 75th percentile",
+        color = ""
+      ) +
+      scale_color_manual(
+        values = c(
+          "Sale" = "darkolivegreen3",
+          "Estimate" = "cadetblue4"
+        )
+      ) +
+      scale_y_continuous(
+        breaks = line_ends$space_maker,
+        labels = line_ends$category
+      ) +
+      scale_x_continuous(
+        labels = label_dollar(),
+        # Prevents last x axis tick mark from being cut off
+        limits = c(NA, max(reshaped_data$value) * 1.04)
+      )
+  )
+}
+```
+
+::: {.panel-tabset}
+
+```{r _model_iterate_sale_vs_estimate_by_class, results='asis', fig.width=8, fig.height=10}
+# Dynamically produce tabset
+for (township in townships_to_iterate) {
+  cat("##", township, "\n")
+  sale_vs_estimate_by_class(township, model_performance_test)
+  cat("\n\n")
+}
+```
+:::
+
```

Comment:
These graphs create a distributional comparison to examine, per township, sale prices vs estimated market values. We look at 25th percentile, median, and 75th percentile with a point representing each. This is also implemented within each township for a class breakout, using a tabset.

This includes both township brea